c10/util/TensorTypeId.cpp

```diff
@@ -1,4 +1,4 @@
-#include "ATen/core/TensorTypeId.h"
+#include "c10/util/TensorTypeId.h"
 #include "caffe2/utils/string_utils.h"
```

Comment:
This include has to go; c10 is not allowed to include caffe2. Probably easiest to move the cluster of functions in caffe2/core/common.h for `to_string` functionality to a new header in util

Reply:
Also, I would only make those changes if they are absolutely necessary. Best is to minimize changes here.

Reply:
@orionr 
@ezyang 

So this change has two questions now
1) Do I need to move TensorTypeId to c10 at all?
2) If yes, then to_string must be removed to separate header.

1) I had to move TensorTypeId to solve unresolved externals problem in Windows with BUILD_SHARED_LIBS=ON caused by recent changes, done in c10 from pytorch/master. You can take this on your own and I revert commit, but not exporting this symbols or reverting problematic c10 changes breaks shared build in Windows.
2) Sure, if this movement of TensorTypeId is possible, I'll create a separate header in c10/util.

Reply:
@ezyang 
@orionr 

Moved the string related ops from common.cuh to c10/util/string_utils.h. If it is the way to go or you will fix windows build on your side anyway its here.